packages/preprocess/src/process-frontmatter/parse-frontmatter.cjs

```diff
@@ -0,0 +1,30 @@
+const fs = require("fs")
+const yaml = require("yaml")
+const frontmatterRegex = /---((?:.|\s)+?)---/
+
+/**
+ * @param {string} content File Content
+ * @returns {any}
+ */
+const parseFrontmatter = (content) => {
+    // Sanity checking
+    // if (!filepath.endsWith(".md")) console.warn("Attempting to extract frontmatter from a non-markdown file")
+    // const stat = fs.statSync(filepath)
+    // if (!stat.isFile()) throw new Error(`${filepath} is not a file`)
+
+    // // Read File
+    // const fileContent = fs.readFileSync(filepath).toString()
+
+    // Run against regex
+    const frontmatterMatches = frontmatterRegex.exec(content)
+
+    // We have a match
+    if (frontmatterMatches?.[1]) {
```

Comment:
Should we have a check on `array` `length`

Reply:
If frontmatter is defined; then we know it is addressable as an array; if you go out of bounds in an array, it comes back as undefined; which fails this conditional:

```
Welcome to Node.js v16.19.1.
Type ".help" for more information.
> let x = []
undefined
> x[1]
undefined
```